### ISSUE TITLE
No Persistence provider for EntityManager named cassandra-pu

### ISSUE THREAD

**Issue by sunrain99** [Impetus/kundera#639]
Hello!
I am using Kundera 2.13 on an dynamic web application to access cassandra database and everything works fine. I created EntityManagerFactory and EntityManager to handle entity bean CRUD. 

this.emf = Persistence.createEntityManagerFactory("cassandra-pu");
EntityManager cem=emf.createEntityManager();

However, after I combine separate web components into EAR package, EntityManagerFactory and EntityManager do not work. Each war package inside EAR has its own persistence.xml and EntityManagerFactory, EntityManager Injection. The error message is:

ERROR [com.impetus.kundera.configure.PersistenceUnitConfiguration](default task-5) Unconfigured persistence unit: cassandra-pu please validate with persistence.xml
ERROR [com.impetus.kundera.KunderaPersistence](default task-5) EntityManagerFactory not created, returning null.
ERROR [org.jboss.as.ejb3.invocation](default task-5) JBAS014134: EJB Invocation failed on component TagManagementBean for method public abstract java.util.List com.sial.enotebook.ejb.TagManagementBeanLocal.viewAll(): javax.ejb.EJBException: javax.persistence.PersistenceException: No Persistence provider for EntityManager named cassandra-pu
.....

Please help and any ideas?

Thanks and best regards,

**Comment by chhavigangwal** [comment]
@sunrain99 

Can you share a sample app to help me debug the issue and resolve the same ?

Chhavi

**Comment by sunrain99** [comment]
@chhavigangwal 
Thanks for the taking a look. How can I share with you my app?

**Comment by sunrain99** [comment]
@chhavigangwal 
I create a war project test_1, its persistence.xml is like this. I can deploy test_1.war to jboss server and run the app successfully.
&lt;persistence-unit name="cassandra-pu">
        &lt;provider>com.impetus.kundera.KunderaPersistence&lt;/provider>  
        &lt;class>entity.Vendor&lt;/class>
        &lt;exclude-unlisted-classes>true&lt;/exclude-unlisted-classes>
        &lt;properties>
            &lt;!-- &lt;property name="jboss.as.jpa.managed" value="false" />-->
            &lt;property name="kundera.name" value="cassandra" />
            &lt;property name="kundera.nodes" value="localhost" />
            &lt;property name="kundera.port" value="9160" />
            &lt;property name="kundera.keyspace" value="project" />
            &lt;property name="kundera.dialect" value="cassandra" />
            &lt;property name="kundera.client" value="thrift" />
            &lt;property name="kundera.client.lookup.class"
                value="com.impetus.client.cassandra.thrift.ThriftClientFactory" />  
        &lt;/properties>
    &lt;/persistence-unit> 

I created another war project test_2, its persistence.xml is like this. I can deploy test_2.war to jboss server and run successfully.
&lt;persistence-unit name="cassandra-pu">
        &lt;provider>com.impetus.kundera.KunderaPersistence&lt;/provider>
        &lt;class>entity.Employee&lt;/class>
        &lt;class>entity.Project&lt;/class>  
        &lt;exclude-unlisted-classes>true&lt;/exclude-unlisted-classes>
        &lt;properties>
            &lt;!--property name="jboss.as.jpa.managed" value="false" / -->
            &lt;property name="kundera.name" value="cassandra" />
            &lt;property name="kundera.nodes" value="localhost" />
            &lt;property name="kundera.port" value="9160" />
            &lt;property name="kundera.keyspace" value="project" />
            &lt;property name="kundera.dialect" value="cassandra" />
            &lt;property name="kundera.client" value="thrift" />
            &lt;property name="kundera.client.lookup.class"
                value="com.impetus.client.cassandra.thrift.ThriftClientFactory" />  
        &lt;/properties>
    &lt;/persistence-unit>

entitymanager is initialized through @CassandraDatabase
        @Inject
    @CassandraDatabase 
    EntityManager entityManager;

public KunderaEntityManagerCDI(){
        System.out.println("will create the entitymanager factory");
        this.emf = Persistence.createEntityManagerFactory("cassandra-pu");
    }
    @Produces @RequestScoped @CassandraDatabase EntityManager getKunderaEntityManager(){

```
    System.out.println("emf is: " + emf);       
    EntityManager cem=emf.createEntityManager();
    cem.setProperty(CassandraConstants.CQL_VERSION, CassandraConstants.CQL_VERSION_3_0);
    return cem;

}
```

Then I created EAR project test and include project test_1 and test_2. test.ear can be deployed to jboss server ok but when I open pages, it got the above errors.

**Comment by chhavigangwal** [comment]
@sunrain99 

I am trying to replicate the same scenario at my end, if possible please share your app by creating a repo over github or email it on kundera@impetus.co.in 

Chhavi

**Comment by sunrain99** [comment]
@chhavigangwal 
I have send you zipped package to kundera@impetus.co.in. Please check it. 
Thanks!

**Comment by chhavigangwal** [comment]
@sunrain99 

Your issue seems to be related to setting classpath in Maifest. Take a look at the relevant thread.     

https://groups.google.com/forum/#!msg/kundera-discuss/3-iUJQOShvE/kJihv32IbjYJ

I will have a look into your project and update you on same.

Chhavi

**Comment by sunrain99** [comment]
@chhavigangwal 
Thanks for your suggestion. I tried but still does not work. Can you please advise again? Thank you and appreciate!
What I did is:
1. in pom.xml, <build><plugin> add
&lt;configuration>  
                    &lt;archive>
                        &lt;manifest>
                          &lt;addClasspath>true&lt;/addClasspath>
                        &lt;/manifest>
                    &lt;/archive>
                &lt;/configuration>
so in project1 and project2 war file, manifest list all class path jar files 
2. add kundera-core-2.13-SNAPSHOT.jar,kundera-cassandra-2.13-SNAPSHOT.jar,kundera-cassandra-ds-driver-2.13-SNAPSHOT.jar,kundera-cassandra-pelops-2.13-SNAPSHOT.jar,fallback-impl-2.13-SNAPSHOT.jar into project\EarContent\lib folder.

**Comment by chhavigangwal** [comment]
Mailed you the working version of ear deployment. Please verify and let me know.

Chhavi

**Comment by chhavigangwal** [comment]
The same deployment works with wildfly 8.1. Closing this one as the same will be resolved with  #534 .

Chhavi
